poc/map/Jack.tbasic

```diff
@@ -0,0 +1,719 @@
+'        Next
+'        
+'    	Case %MOSTLYWIDE
+'    	
+'        wideTotal = 0
+'        
+'        for j as integer = 1 to me.worldList(i).ambiance.riverArmsCount
+'          'Width
+'          if j = me.worldList(i).ambiance.riverArmsCount Then
+'            me.worldList(i).arms(j).width += iseven(assign(me.worldList(i).minimumTotalRiverWidth - wideTotal, me.worldList(i).arms(j).width))
+'          Else
+'            me.worldList(i).arms(j).width += iseven(assign(ceil(me.worldList(i).minimumTotalRiverWidth / (me.worldList(i).ambiance.riverArmsCount + j - 1)), me.worldList(i).arms(j).width))
+'            wideTotal += me.worldList(i).arms(j).width
+'          endif
+'          
+'          me.worldList(i).widestArmWidth = max(me.worldList(i).widestArmWidth, me.worldList(i).arms(j).width)
+'          
+'          'Random depth profile
+'          me.worldList(i).arms(j).depthProfile = getRandomFlag(me.worldList(i).ambiance.armsDepthProfiles)
+'        Next
+'        
+'    End Select
+'    
+'    '[todo] <Something to do here>
+'    ' Here sorting me.worldList(i).arms(j).width wide to narrow ; how array sort ?
+'   #Region " Sort arms by width descending"
+'    Dim sorted as Boolean
+'    DIm tempArm as tArm
+'      '[breakpoint] <Any note here. Breakpoint will be set to line following this statement>
+'    Do
+'      sorted = %TRUE
+'      for j = 2 to me.worldList(i).ambiance.riverArmsCount
+'        if me.worldList(i).arms(j).width > me.worldList(i).arms(j-1).width Then
+'        	sorted = %False
+'        	tempArm = me.worldList(i).arms(j).width
+'        	me.worldList(i).arms(j-1).width = me.worldList(i).arms(j).width
+'        	me.worldList(i).arms(j).width = tempArm
+'        End If
+'      Next
+'    loop until sorted = %TRUE
+'   #EndRegion 
+'  Next
+'  
+#EndRegion
+
+
+#Region "Workaround version River properties"
+
+  for i as Integer = 1 to me.worldCount
+    
+    If i < me.worldCount Then
+      me.worldList(i).streamsOutCount = min(me.worldList(i+1).ambiance.riverArmsCount, me.worldList(i).ambiance.riverArmsCount)
+    endif
+    
+    if me.worldList(i).ambiance.riverArmsCount = 1 then me.worldList(i).ambiance.riverArmWidthDistribution = %EQUAL
+  
+    select case me.worldList(i).ambiance.riverArmWidthDistribution
+    
+    	Case %EQUAL
+    	
+        if me.worldList(i).ambiance.riverArmsCount = 1 Then
+          equalWidth = me.worldList(i).minimumTotalRiverWidth
+        else
+          equalWidth = 1 + 2 * Ceil(((me.worldList(i).minimumTotalRiverWidth - me.worldList(i).ambiance.riverArmsCount)/me.worldList(i).ambiance.riverArmsCount)/2)
+        endif
+        me.worldList(i).widestArmWidth = equalWidth
+        for j as integer = 1 to me.worldList(i).ambiance.riverArmsCount
+          'Width
+          me.worldList(i).arms(j).width = equalWidth
+          'Random depth profile
+          me.worldList(i).arms(j).depthProfile = getRandomFlag(me.worldList(i).ambiance.armsDepthProfiles)
+        Next
+        
```

Comment:
Here after the for/next j is still defined. Why ?

Reply:
Well, it is a bit controversial feature in thinBasic. The declared variable does not live in FOR/NEXT scope, but stays aftewards.